File: java/com/aa/act/interview/org/MyOrganization.java
package com.aa.act.interview.org;

public class MyOrganization extends Organization {

	@Override
	protected Position createOrganization() {
		Position ceo = new Position("CEO");
		Position pres = new Position("President");
		ceo.addDirectReport(pres);
		Position vpm = new Position("VP Marketing");
		pres.addDirectReport(vpm);
		Position vps = new Position("VP Sales");
		pres.addDirectReport(vps);
		Position vpf = new Position("VP Finance");
		pres.addDirectReport(vpf);
		Position coo = new Position("COO");
		pres.addDirectReport(coo);
		Position cio = new Position("CIO");
		ceo.addDirectReport(cio);
		Position vpt = new Position("VP Technology");
		cio.addDirectReport(vpt);
		Position vpi = new Position("VP Infrastructure");
		cio.addDirectReport(vpi);
		Position dea = new Position("Director Enterprise Architecture");
		vpt.addDirectReport(dea);
		Position dct = new Position("Director Customer Technology");
		vpt.addDirectReport(dct);
		Position s = new Position("Salesperson");
		vps.addDirectReport(s);
		
		return ceo;
	}

	public static final void main(String... args) {
		MyOrganization myOrg = new MyOrganization();
		myOrg.hire(new Name("Doug", "Parker"), "CEO");
		myOrg.hire(new Name("Robert", "Isom"), "President");
		myOrg.hire(new Name("Gandalf", "Gray"), "Director Enterprise Architecture");
		myOrg.hire(new Name("Head", "Geek"), "Director Customer Technology");
		myOrg.hire(new Name("Jane", "Smith"), "VP Marketing");
		myOrg.hire(new Name("Jim", "Jones"), "VP Sales");
		myOrg.hire(new Name("Bean", "Counter"), "VP Finance");
		myOrg.hire(new Name("Maya", "Liebman"), "CIO");
		myOrg.hire(new Name("Danielle", "Hoover"), "VP Technology");
		myOrg.hire(new Name("Scape", "Goat"), "VP Infrastructure");
		myOrg.hire(new Name("Slick", "Willie"), "Salesperson");
		System.out.println(myOrg);
	}
}

File: javascript-es5/MyOrganization.js
const Name = require('./Name.js');
const Organization = require('./Organization.js');
const Position = require('./Position.js');

const ceo = Position('CEO');
const pres = Position('President');
ceo.addDirectReport(pres);
const vpm = Position('VP Marketing');
pres.addDirectReport(vpm);
const vps = Position('VP Sales');
pres.addDirectReport(vps);
const vpf = Position('VP Finance');
pres.addDirectReport(vpf);
const coo = Position('COO');
pres.addDirectReport(coo);
const cio = Position('CIO');
ceo.addDirectReport(cio);
const vpt = Position('VP Technology');
cio.addDirectReport(vpt);
const vpi = Position('VP Infrastructure');
cio.addDirectReport(vpi);
const dea = Position('Director Enterprise Architecture');
vpt.addDirectReport(dea);
const dct = Position('Director Customer Technology');
vpt.addDirectReport(dct);
const s = Position('Salesperson');
vps.addDirectReport(s);

const myOrg = Organization(ceo);

myOrg.hire(Name('Bob', 'Smith'), 'CEO');
myOrg.hire(Name('Zaphod', 'Beeblebrox'), 'President');
myOrg.hire(Name('Gandalf', 'Gray'), 'Director Enterprise Architecture');
myOrg.hire(Name('Bill', 'Lumbergh'), 'Director Customer Technology');
myOrg.hire(Name('Ford', 'Prefect'), 'VP Marketing');
myOrg.hire(Name('Jane', 'Seller'), 'VP Sales');
myOrg.hire(Name('Bean', 'Counter'), 'VP Finance');
myOrg.hire(Name('Victoria', 'Sinclair'), 'CIO');
myOrg.hire(Name('Head', 'Geek'), 'VP Technology');
myOrg.hire(Name('Steve', 'Dent'), 'VP Infrastructure');
myOrg.hire(Name('Slick', 'Willie'), 'Salesperson');

console.log(myOrg.toString());
